### DISCUSSION TITLE
Suggestion to improve Attachments API

### DISCUSSION THREAD

**Discussion by georgiosd** [ravendb/ravendb#16226] [Ideas]
I just got "caught" by a subtle thing here:

```
async Task Main()
{
	var store = new DocumentStore
	{
		Urls = new[] { "http://localhost:8080" },
		Database = "test"
	};
	
	store.Initialize();
	
	using (var session = store.OpenAsyncSession())
	{
		await session.StoreAsync(new object(), "tests/5");
		
		await session.SaveChangesAsync();
	}
	
	using (var session = store.OpenAsyncSession())
	{
		var e = await session.Advanced.Attachments.GetAsync(new[]
		{
			new AttachmentRequest("tests/5", "test.pdf")
		});
		
		foreach(var attachment in e)
		{
			e.Dump();
		}
	}
}
```

This code will print nothing, because asking for an attachment that doesn't exist will simply not return a value in the enumerator.

I would like to propose either an optional (and backwards compatible parameter) that will throw if the attachment is not found, or a new method that does this.

Happy to submit a PR for it!

**Comment by ppekrol** [comment]
Hi @georgiosd 

I think we can add additional ctor to `GetAttachmentsOperation` and then this operation can be used directly with the new parameter instead of through the session. This way we would not have to worry about the backward compatibility.

We would be happy to accept a PR from you.
